app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,121 @@
+class RecipesController < ApplicationController
+  include ActionView::Helpers::NumberHelper
+
+  before_action :set_recipe, only: %i[show edit update destroy generate_shopping_list]
+
+  def index
+    @recipes = Recipe.all
+  end
+
+  def show
+    @recipe = Recipe.includes(:foods).find(params[:id])
+    @foods = @recipe.foods
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.build(recipe_params)
+
+    if @recipe.save
+      redirect_to @recipe, notice: 'Recipe was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def public_recipes
+    # @public_recipes = Recipe.where(public: true)
+    @public_recipes = Recipe.public_recipes.where.not(user: current_user)
+  end
```

Comment:
- [ ] Please kindly remove this commented out code in order for your work to be neat. ℹ️